src/Symfony/Component/DependencyInjection/Container.php

```diff
@@ -359,14 +358,19 @@ public function initialized($id)
     public function getServiceIds()
     {
         $ids = array();
+        $reversedMethodMap = array_change_key_case(array_flip($this->methodMap), \CASE_LOWER);
```

Comment:
> you cant imagine how creative users can

expect that as you spotted, it never worked :)

> What if we make it feature?

then it should go on master and should have a supporting use case